rfcs/0000_intelligent_resampling.md

```diff
@@ -0,0 +1,368 @@
+
+- **Hypothesis coverage**: If some hypotheses remain valid after prediction errors, the existing models are likely sufficient and we should focus on hypothesis elimination. However, if all hypotheses are eliminated (or less than a certain percentage threshold), we may need to learn a new model or update an existing one.
+
+- **Observation frequency** (e.g., `_observation_count` in `GridObjectModel`): Frequently visited locations with high observation counts suggest the model is well-learned at that location, biasing toward hypothesis elimination rather than model updates.
+
+- **Compositionality**: When prediction errors occur at specific locations on an otherwise well-matched familiar object, consider learning the modification as a compositional element rather than updating the base model. This preserves the integrity of well-learned models while capturing local variations or modifications.
+
+- **Error magnitude thresholds**: Very large prediction errors across all features may suggest a novel object requiring a new model. Moderate errors might indicate the need for model updates or hypothesis refinement.
+
+- **Feature-specific patterns**: If morphological features match but non-morphological features (like color) differ significantly, this might indicate object variations / keyframe / need to separate morphological and feature models.
+
+#### Example of Chipped Coffee Mug 
+
+Consider the TBP mug that we are very familiar that is now slightly chipped. Below are how the above heuristics may apply:
+
+1. **Initial detection**: When sensing the chipped area, we would get prediction errors since the surface geometry differs from our stored model.
+2. **Hypothesis coverage**: The hypothesis for where we are on the mug would remain valid across many locations during movement (handle, sides, rim); it would only result in a prediction error / risk elimination when we move to the chip, the point at which we would want to signal learning in a model.
+3. **Observation frequency**: The mug model has high observation counts from frequent use, suggesting we shouldn't discard the entire model.
+4. **Feature patterns**: Morphological features (surface curvature) would differ at the chip location, but non-morphological features (color, texture) might remain consistent.
+5. **Compositionality**: Since the chip is a localized modification at specific coordinates on an otherwise familiar object, the system could learn it compositionally - preserving the original mug model while learning the chip as a local variation.
+6. **Decision**: The heuristics would guide the system to learn the chipped area as a compositional modification rather than corrupting the well-learned base mug model. This allows us to maintain our knowledge about the standard mug shape while capturing this specific instance's variation. The lower-level mug model remains intact and reusable, while the chip information is stored as a localized modification.
```

Comment:
This seems to go into a new direction from the original question of this RFC ("How do we update a model if the object changed?" vs. "How can we use prediction errors to eliminate hypotheses when the sensor has moved off of an object? ". I would give an example here that tackles that specific question you set out with.